pywbem/_listener.py

```diff
@@ -1146,10 +1164,54 @@ def stop(self):
             self._https_thread = None
             self.logger.info("Stopped threaded HTTPS server")
 
-    def deliver_indication(self, indication, host):
+        self.logger.info("Stopping queued delivery")
+        self.delivery_stop = True
+        self.logger.info("Stopped threaded Deliver queue")
+
+    def deliver_indications_from_queue(self, delivery_queue):
+        """
+        Deliver indications from delivery_queue to the defined consumer. This
+        function runs in its own thread and only stops when the listener is
+        shut down.
+
+        """
+        self.logger.debug("Deliver from queue.")
+        # If indications in timeout period, deliver the indication.
+        delivery_loop = True
+        while delivery_loop:
+            try:
+                self.logger.debug("Deliver from queue.")
+                indication_tuple = delivery_queue.get(block=True, timeout=30)
+                self.deliver_indication_to_callback(indication_tuple[0],
+                                                    indication_tuple[1])
+            # FUTURE: Should we be able to turn off this thread if there are
+            # no indications received for a defined time? Now just continues
+            # the receive process
+            except queue.Empty:
+                if self.delivery_stop:
+                    delivery_loop = False
+        self.logger.debug("Return from deliver_indications_from_queue")
+        return
+
+    def handle_indication(self, indication, host):
+        """
+        Entry point from the listener with a single indication.
+        Either deliver to callback or put in interprocess queue
+        """
+        # Test code to allow testing with bypass of the queued return.
+        # Set this to False to test with direct callback.
+        queued_delivery = True
```

Comment:
I'm not sure I can relate this comment to the code in case of False.
In addition, I think we should have only one way implemented, for simplicity.

Reply:
Removed the direct callback and yes, it was wrong